storm-core/src/jvm/backtype/storm/windowing/EvictionPolicy.java

```diff
@@ -39,4 +63,12 @@
      * @param event the input event to be tracked
      */
     void track(Event<T> event);
+
+    /**
+     * Sets a context in the eviction policy that can be used while evicting the events.
+     * E.g. For TimeEvictionPolicy, this could be used to set the reference timestamp.
+     *
+     * @param context
+     */
+    void setContext(Object context);
```

Comment:
May want to introduce class/interface EvictionPolicyContext which may have opaque structure for now. This can be done later as this PR is waiting for long.